src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -0,0 +1,304 @@
+package bio.terra.service.snapshotbuilder.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToCompressingWhiteSpace;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.grammar.azure.SynapseVisitor;
+import bio.terra.model.SnapshotBuilderCriteria;
+import bio.terra.model.SnapshotBuilderCriteriaGroup;
+import bio.terra.model.SnapshotBuilderDomainCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataListCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+class CriteriaQueryBuilderTest {
+
+  @Test
+  void generateRangeCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataRangeCriteria rangeCriteria = generateRangeCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder("person", SynapseVisitor.azureTableName("source_dataset_name"))
+            .generateFilterForRangeCriteria(rangeCriteria);
+
+    // Table name is null because there is no alias generated until it is rendered as a full query
+    assertThat(
+        "The sql generated is correct",
+        filterVariable.renderSQL(),
+        equalToCompressingWhiteSpace(
+            "(null.range_column_name >= 0 AND null.range_column_name <= 100)"));
```

Comment:
Testing by rendering to test duplicates all the tests that already exist which verify that `renderSql()` works correctly. It works but it depend on code that's not part of the class being tested; a test which didn't do that would instead compare the objects directly. Right now this is awkward to do since our sql expression object don't support `equals()`, so you'd have to compare the objects field-by-field. It's probably OK for now to leave it using render sql but at some point it might be nice to change this.